src/main/java/teammates/ui/webapi/CalculateUsageStatisticsAction.java

```diff
@@ -0,0 +1,88 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+import teammates.common.datatransfer.QueryLogsResults;
+import teammates.common.datatransfer.attributes.UsageStatisticsAttributes;
+import teammates.common.datatransfer.logs.GeneralLogEntry;
+import teammates.common.datatransfer.logs.LogEvent;
+import teammates.common.datatransfer.logs.LogSeverity;
+import teammates.common.datatransfer.logs.QueryLogsParams;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Logger;
+import teammates.common.util.TimeHelper;
+
+/**
+ * Gathers usage-related statistics (e.g. new created entities) in the past hour and store in the database.'
+ */
+public class CalculateUsageStatisticsAction extends AdminOnlyAction {
+
+    static final int COLLECTION_TIME_PERIOD = 60; // represents one hour
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    public JsonResult execute() {
+        Instant endTime = TimeHelper.getInstantNearestHourBefore(Instant.now());
+        Instant startTime = endTime.minus(COLLECTION_TIME_PERIOD, ChronoUnit.MINUTES);
+
+        UsageStatisticsAttributes entitiesStats = logic.calculateEntitiesStatisticsForTimeRange(startTime, endTime);
+        int numEmailsSent = getNumberOfLogsForEvent(startTime, endTime, EventType.EMAIL_SENT);
+        int numSubmissions = getNumberOfLogsForEvent(startTime, endTime, EventType.SUBMISSION);
+
+        UsageStatisticsAttributes overallUsageStats = UsageStatisticsAttributes.builder(startTime, COLLECTION_TIME_PERIOD)
+                .withNumResponses(entitiesStats.getNumResponses())
+                .withNumCourses(entitiesStats.getNumCourses())
+                .withNumStudents(entitiesStats.getNumStudents())
+                .withNumInstructors(entitiesStats.getNumInstructors())
+                .withNumAccountRequests(entitiesStats.getNumAccountRequests())
+                .withNumEmails(numEmailsSent)
+                .withNumSubmissions(numSubmissions)
+                .build();
+
+        try {
+            logic.createUsageStatistics(overallUsageStats);
+        } catch (InvalidParametersException | EntityAlreadyExistsException e) {
+            log.severe("Unexpected error", e);
+        }
+        return new JsonResult("Successful");
+    }
+
+    private int getNumberOfLogsForEvent(Instant startTime, Instant endTime, EventType eventType) {
+        int total = 0;
+        long logStartTime = startTime.toEpochMilli();
+        LogEvent logEvent;
+        String extraFilters = "";
+        if (eventType == EventType.SUBMISSION) {
+            logEvent = LogEvent.FEEDBACK_SESSION_AUDIT;
+            extraFilters = "jsonPayload.accessType=\"submission\"";
+        } else {
+            logEvent = LogEvent.EMAIL_SENT;
+        }
+        while (true) {
```

Comment:
Would this loop be better abstracted into a method to get the number of entries? It may be too low-level and have alot of query-related logic to be placed in an action

Reply:
In the end I moved everything from here to `LogsProcessor`. The method is rather niche right now (only accepts `LogEvent` and extra filters on top of the two timestamps), but I guess it's not a big deal.